symengine/polynomial.cpp

```diff
@@ -208,34 +208,34 @@ bool UnivariateIntPolynomial::is_pow() const {
     return false;
 }
 
-RCP<const UnivariateIntPolynomial> add_poly(const UnivariateIntPolynomial &a, const UnivariateIntPolynomial &b) {
+RCP<const UnivariateIntPolynomial> add_poly(const RCP<const UnivariateIntPolynomial> &a, const RCP<const UnivariateIntPolynomial> &b) {
```

Comment:
The exact reason for this is here https://github.com/symengine/symengine/pull/406#issuecomment-77592690

Sorry I couldn't find the exact comment earlier.